content/blog/scale-kubernetes-clusters-using-hpe-greenlake-terraform-provider.md

```diff
@@ -0,0 +1,394 @@
+2. Upgrade the kubernetes version of the cluster. Covered in this blog.
+3. Upgrade the OS version of the worker nodes inside a cluster. Covered in this blog.
+
+# Prerequisite
+
+Before starting this tutorial, it is recommended that you read the blog post  [Kubernetes Cluster as Code - Part 1](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), which includes steps for creating a Kubernetes cluster. This post expands upon that scenario by examining how to upgrade a cluster's kubernetes version & OS version of its woker nodes. 
+
+# Verify existing Kubernetes cluster
+
+After the cluster is created following the instructions found in the  [Kubernetes Cluster as Code - Part 1 blog post](https://developer.hpe.com/blog/kubernetes-clusters-as-code-part1/), launch HPE GreenLake Central console and verify that the cluster is present under the appropriate tenant.
+
+You should see the  **tf-test**  cluster present under  **Dashboard -> Manage your Private Cloud -> Containers**.
+
+![](https://developer.hpe.com/img/cluster_list.jpg)
+
+Below is the reference Terraform configuration file for the existing cluster.
+
+```hcl
+terraform {
+  required_providers {
+    hpegl = {
+      source  = "hpe/hpegl"
+      version = ">= 0.2.2"
+    }
+  }
+}
+ 
+provider hpegl {
+  caas {
+  }
+}
+ 
+variable "HPEGL_SPACE" {
+  type = string
+}
+ 
+data "hpegl_caas_site" "blr" {
+  name = "BLR"
+  space_id = var.HPEGL_SPACE
+ }
+ 
+data "hpegl_caas_cluster_blueprint" "bp" {
+  name = "demo"
+  site_id = data.hpegl_caas_site.blr.id
+}
+ 
+resource hpegl_caas_cluster test {
+  name         = "tf-test"
+  blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+  site_id      = data.hpegl_caas_site.blr.id
+  space_id     = var.HPEGL_SPACE
+}
+```
+
+# Kubernetes version upgrade
+
+For Kubernetes version upgrade, we need to specify the new version of kubernetes that is available for upgrade in the resources block. 
+
+1. **kubernetes_version**: *Use the Kubernetes version that pops up on the cluster details page in the UI.*
+
+   ![](/img/blog1.jpg)
+
+
+
+ Below is the reference Terraform configuration for updating the cluster's kubernetes version.
+
+```hcl
+terraform {
+ required_providers {
+   hpegl = {
+     source  = "hpe/hpegl"
+     version = ">= 0.2.2"
+   }
+ }
+}
+
+provider hpegl {
+ caas {
+ }
+}
+
+variable "HPEGL_SPACE" {
+ type = string
+}
+
+data "hpegl_caas_site" "blr" {
+ name = "BLR"
+ space_id = var.HPEGL_SPACE
+}
+
+data "hpegl_caas_cluster_blueprint" "bp" {
+ name = "demo"
+ site_id = data.hpegl_caas_site.blr.id
+}
+
+resource hpegl_caas_cluster test {
+ name         = "tf-test"
+ blueprint_id = data.hpegl_caas_cluster_blueprint.bp.id
+ site_id      = data.hpegl_caas_site.blr.id
+ space_id     = var.HPEGL_SPACE
+ kubernetes_version = "1.23.13-hpe1"
+}
+```
+
+# Ready to Terraform plan
```

Comment:
Suggest using active voice: "Run Terraform plan"

Reply:
Suggest to make it a level 2 header.